cluster-autoscaler/simulator/scheduling/hinting_simulator.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scheduling
+
+import (
+	"fmt"
+
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/clustersnapshot"
+	"k8s.io/autoscaler/cluster-autoscaler/simulator/predicatechecker"
+
+	apiv1 "k8s.io/api/core/v1"
+	klog "k8s.io/klog/v2"
+	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+// HintingSimulator is a helper object for simulating scheduler behavior.
+// TODO(x13n): Reuse this in filter_out_unschedulable.
```

Comment:
There are issue templates for bug reports and feature requests, this would be neither one. I see https://github.com/kubernetes/autoscaler/pull/5259 is refactoring filter_out_unschedulable already though, so maybe it would make sense to make this change there? @BigDarkClown

Reply:
I think #5259 is already pretty big as is, so let's not merge the two.